h2d/Particles.hx

```diff
@@ -351,6 +356,21 @@ class ParticleGroup {
 		p.vy *= speed;
 		p.life = 0;
 		p.maxLife = life;
+
+		if ( !isRelative ) {
+			// Less this.parts access
+			var parts = this.parts;
+			// calcAbsPos() was already called, because during both rebuild() and Particle.update()
+			// called during sync() call which calls this function if required before any of this happens.
+			//parts.syncPos();
+
+			p.x = p.x * parts.matA + p.y * parts.matC + parts.absX;
+			p.y = p.x * parts.matB + p.y * parts.matD + parts.absY;
+			p.scaleX = Math.sqrt((parts.matA * parts.matA) + (parts.matC * parts.matC)) * size;
+			p.scaleY = Math.sqrt((parts.matB * parts.matB) + (parts.matD * parts.matD)) * size;
+			p.rotation += Math.acos(parts.matA / p.scaleX);

```

Comment:
I'm still a bit unsure about that Math.acos here. I think an Math.atan2(matB / scaleY, matA / scaleX) would be better but I haven't tested it.

Reply:
Yes, you're right (it does back-and-forth movement which I couldn't notice at first), also noticed another bug with px/py calcualation. Due to `p.x` being assigned first, it's new value affects `p.y`, while it supposed to use old one. Plus I overlooked the need to rotate movement velocity as well.